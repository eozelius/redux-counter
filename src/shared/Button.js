import React, { Component } from 'react'

class Button extends Component {
  render(){

    return (
      <button onClick={this.props.handleClick} style={{ margin: '5%' }}>
        {/* this.props.children */}
        { this.props.children ? this.props.children : 'click'}
      </button>
    )
  }
}

export default Button
