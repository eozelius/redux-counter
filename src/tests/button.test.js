import React from 'react'
import Button from '../shared/Button'

describe('Button', () => {
  it('should render correctly', () => {
    const btn = shallow(<Button />)
    expect(btn).toMatchSnapshot()
  })

  it('should render the child text', () => {
    const btn = shallow(<Button>testing 123</Button>)
    expect(btn.prop('children')).toEqual('testing 123')
  })

  it('defaults to "click"', () => {
    const btn = shallow(<Button />)
    expect(btn.prop('children')).toEqual('click')
  })

  it('invokes the handleClick callback passed via props', () => {
    const handleClick = jest.fn()
    const btn = shallow(<Button handleClick={handleClick} />)
    btn.simulate('click', {})
    expect(handleClick).toBeCalledWith({})
  })
})
