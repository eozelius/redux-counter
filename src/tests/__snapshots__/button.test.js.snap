// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  style={
    Object {
      "margin": "5%",
    }
  }
>
  click
</button>
`;
